### ISSUE TITLE
Stream remove label ?

### ISSUE THREAD

**Issue by TarekGharsalli** [pinellolab/STREAM#130]
Hello, 

Does any one encounter this problem when he try to add label to the adata and he don't find all the labels when he plot ?
I have 8 clusters and I added the label to my adata (cl0, cl1, .... cl8) and when I plot for example st.plot_dimension_reduction, I don't found cl0 in the legend. I use another label and the same thing, there is usually one cluster lacking. 

Thanks

**Comment by huidongchen** [comment]
Hi , would you mind sharing with me a dummy data to reproduce the error so that I can look into the issue?

**Comment by TarekGharsalli** [comment]
HI @huidongchen ,

Thanks for your reply,
I can share the dataset here ?
I can show you here an images to expalain more  
I load the 10X data, then I add the labels; as you see in the second image that I have a  Cluster_0, then I plot the dimension reduction colored by label and I don't found Cluster_0, and when I check again in the the adata , I found label Cluster_0.
![fig1](https://user-images.githubusercontent.com/60474281/143216690-991dda2b-2f8d-4cf7-9e51-9c4c9229175c.png)
![fig2](https://user-images.githubusercontent.com/60474281/143216691-0d7ae04d-0b14-4c3c-997a-8040f5ca14ce.png)
![fig3](https://user-images.githubusercontent.com/60474281/143216693-a1f5a534-7611-4674-ada8-10781873253c.png)

**Comment by huidongchen** [comment]
hi thanks for sharing the details. 

Unfortunately given this information, I can not really tell what's going on. Can you save the `adata` to a `h5ad` file (make sure to run `st.write(adata, file_name='adata.h5ad')` before the step `st.dimension_reduction()` )) so that i can try to reproduce the issue myself?

**Comment by TarekGharsalli** [comment]
Hi @huidongchen 

Where can I share the datatset ? 

Thanks !

**Comment by huidongchen** [comment]
Can you upload the dataset here if it is just a dummy dataset?

**Comment by wong-ziyi** [comment]
I am experiencing the same issue.
![image](https://user-images.githubusercontent.com/9590738/145666733-54749e6c-f9bb-4895-bfc0-31c180021335.png)

**Comment by wong-ziyi** [comment]
Hi,
I just solved this problem by doing the things in issue #122 
Please open the file named core.py in the directory of your stream package and replace the code `ax_i.legend(handles=legend_handles[1:], labels=legend_labels[1:]` to `ax_i.legend(handles=legend_handles[0:], labels=legend_labels[0:]` in the line # 1414, 1467, 2780, and 2970.

My stream version = 1.0

![image](https://user-images.githubusercontent.com/9590738/145667011-466624bc-bd18-43c6-bd75-ee065599aa70.png)

**Comment by TarekGharsalli** [comment]
hi @wong-ziyi 

Thanks for your reply 
Where I can find core.py ? 
Thanks

**Comment by TarekGharsalli** [comment]
Hello @huidongchen ,

I tried to load the dataset here but it's doesn't work

**Comment by huidongchen** [comment]
It is actually an issue caused by the latest version of `seaborn` in which the legend object has been changed. 

I have pushed several fixes. If you install the latest version from the master branch (within your current env), it should fix the issue.

```bash
git clone https://github.com/pinellolab/STREAM.git
cd STREAM
pip install .
```

I will try to get it updated in the Bioconda channel as well soon.

**Comment by huidongchen** [comment]
The issue has been fixed in `stream v1.1`.
